packages/jaeger-ui/src/components/common/SearchableSelect.test.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { Select } from 'antd';
+import SearchableSelect, { filterOptionsByLabel } from './SearchableSelect';
+
+describe('SearchableSelect', () => {
```

Comment:
You're missing the test of actually searching when typing. Instead of testing the internal findBy... function, you could've made those same tests from the perspective of the end user, who is typing into the field, not calling a function.

Reply:
Yes, that's how we initially planned to do it but isn't the part for showing matching options on the UI handled by Ant Design that is already tested? I can add an explicit UI test if that makes more sense

Reply:
Recently we broke the ability to search and no unit tests failed. If something changes in ant design regarding HOW filtering is supposed to be hooked up, no tests will fail again. So (a) we need to test for that, and (b) why restrict the input permission tests that you have to test inner function if we could test them end to end to the user facing behavior?

Reply:
Sure, that makes more sense